scrowl-io/owl-ui
# [sc-352] - Set up lint and prettier scripts

### Short summary
- Set up Prettier tasks to check and fix code formatting.
- Set up ESLint tasks to analyze and quick fix issues in js files

The following commands were added to the `package.json`:
```
    "lint:fix": "yarn lint:fix:eslint && yarn lint:fix:prettier"
    "lint:fix:eslint": "eslint --fix \"packages/**/*.{ts,tsx,js,jsx}\""
    "lint:fix:prettier": "prettier --write \"packages/**/*.{json,ts,tsx,js,jsx}\""
    "lint:check": "yarn lint:check:eslint && yarn lint:check:prettier"
    "lint:check:eslint": "eslint \"packages/**/*.{ts,tsx,js,jsx}\""
    "lint:check:prettier": "prettier --check \"packages/**/*.{json,ts,tsx,js,jsx}\""
```
This is going to automate code fixing and formatting, so before getting a PR into review, the dev may run `yarn lint:check` to check out the style and coding issues, and then `yarn lint:fix` to fix all of them manually.

---
### Test steps
Run the following CLI commands to execute the tasks and ensure you don't get any errors besides the expected Prettier and ESLint ones:
1. `yarn lint:check:prettier` - you may see a list of files where style issues were found
2. `yarn lint:check:eslint` - you may see a list of js coding issues, such as missing trailing commas, semi-colons, and unnecessary empty lines
3. `yarn lint:check` - it runs both commands above
4. `yarn lint:fix:prettier` - Prettier is going to fix the style issues it found when you ran `lint:check:prettier`. Some issues may have to be fixed manually. You can see all changes clicking on the `Source Control` icon on the sidebar, it will show a list of changed files.
5. `yarn lint:fix:eslint` - ESLint will fix javascript coding, things like changing `let` to `const`, add `( )` in component returns. Most changes will be made by Prettier.
6. `yarn lint:fix` - it runs bot commands above altogether
